### ISSUE TITLE
LibraryGlideModule can not run independently.

### ISSUE THREAD

**Issue by shimika** [bumptech/glide#4657]
**4.12.0**:

**Glide, Interceptor, Compile (for annotationProcessor)**:

**Emulator**:

**Issue details / Repro steps / Use case background**: 

I make a library use glide to show images on chat.
I should to use GlideModule to add header when request image.
So, I make a LibraryGlideModule and add OkhttpClient in registerComponents function.
But it not called.

After add AppGlideModule in other module or library module (I test two cases each)
It works and LibraryGlideModule's registerComponents function is called.

In glide module
1. We should to use LibraryGlideModule except AppGlideModule in library.
2. LibraryGlideModule won't work without AppGlideModule.

In conclusion, We should guide to developers who use our library to add empty AppGlideModule implementation
It is very annoying if someone not use image viewer.

What is good practice in this case?

**Comment by justziyang** [comment]
hi，have you resolved it?
